llama_stack/apis/scoring_functions/scoring_functions.py

```diff
@@ -135,15 +245,25 @@ class ScoringFunctions(Protocol):
     async def list_scoring_functions(self) -> ListScoringFunctionsResponse: ...
 
     @webmethod(route="/scoring-functions/{scoring_fn_id:path}", method="GET")
-    async def get_scoring_function(self, scoring_fn_id: str, /) -> Optional[ScoringFn]: ...
+    async def get_scoring_function(
+        self,
+        scoring_fn_id: str,
+    ) -> Optional[ScoringFn]: ...
 
     @webmethod(route="/scoring-functions", method="POST")
     async def register_scoring_function(
         self,
-        scoring_fn_id: str,
-        description: str,
-        return_type: ParamType,
-        provider_scoring_fn_id: Optional[str] = None,
-        provider_id: Optional[str] = None,
-        params: Optional[ScoringFnParams] = None,
-    ) -> None: ...
+        fn: ScoringFnParams,
+        scoring_fn_id: Optional[str] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+    ) -> ScoringFn:
+        """
+        Register a new scoring function with given parameters.
+        Only valid scoring function type that can be parameterized can be registered.
+
+        :param fn: The type and parameters for the scoring function.
+        :param scoring_fn_id: (Optional) The ID of the scoring function to register. If not provided, a random ID will be generated.
+        :param metadata: (Optional) Any additional metadata to be associated with the scoring function.
+            - E.g. {"description": "This scoring function is used for ..."}
+        """
+        ...
```

Comment:
class ScoringFunctionTypeInfo:
     type:
     examples: str
     description: str
     supported_purpose: List[str]

Reply:
added list endpoint for getting available types